### PR TITLE
OCM-3831 | fix: 'rosa describe admin' cannot find 'cluster-admin' htpasswd idp and user

### PR DESCRIPTION
**Pull request by davidleerh**
Previously, `rosa describe admin` was not able to locate the correct IDP that contains information about the admin user associated with the cluster.

Setup (manually validate that cluster has an admin user defined under idp)
Example Cluster: 26bo9uvb0csf5r8vn5n76jeotpabpb2n (dle-0919)
image

Before fix:
Running `rosa describe admin -c 26bo9uvb0csf5r8vn5n76jeotpabpb2n`
image

After fix:
image

### CODE REVIEW

**Comment by ckandag** [comment]
@davidleerh   In the screenshots in PR description above - How was the admin user created ?  Was it through the cluster-create-time cluster-admin flag ? or through rosa create admin ?

**Comment by davidleerh** [comment]
> @davidleerh In the screenshots in PR description above - How was the admin user created ? Was it through the cluster-create-time cluster-admin flag ? or through rosa create admin ?

Hi @ckandag thank you for taking a look!
The admin was created using rosa create admin via the interactive mode (example below):
image

File: cmd/create/admin/cmd.go [hunk 1]
```diff
@@ -221,33 +221,20 @@ func FindExistingClusterAdminIDP(cluster *cmv1.Cluster, r *rosa.Runtime) (
 		os.Exit(1)
 	}
 
+	// Locates IDP with name "cluster-admin" that stores the admin user(s)
 	for _, item := range idps {
-		if ocm.IdentityProviderType(item) == ocm.HTPasswdIDPType {
-
+		idpName, idpNameExists := item.GetName()
```
Comment (marcolan018) [inline cmd/create/admin/cmd.go:226]: do we have a case that idpNameExists is False?

**Comment by ckandag** [comment]
> > @davidleerh In the screenshots in PR description above - How was the admin user created ? Was it through the cluster-create-time cluster-admin flag ? or through rosa create admin ?
> 
> Hi @ckandag thank you for taking a look! The admin was created using rosa create admin via the interactive mode (example below): image

The screenshot shows `rosa cluster create`    not `rosa create amin`

File: cmd/create/admin/cmd.go (hunk 1 above)
Comment (davidleerh) [reply cmd/create/admin/cmd.go:226]: I don't believe the name of the IDP will ever be empty as there are validations built in on both the CLI and the backend . Should the check be removed?

**Comment by davidleerh** [comment]
Thank you for your suggestions @ckandag 

As a follow up,
further manual tests were done to investigate the issue further.

With the command, `rosa create admin`
It automatically creates an admin user with the username "cluster-admin" under the htpasswd idp that is also named "cluster-admin"
(IDP NAME)
image


(USERNAME) image

With the command "rosa create cluster via flag and/or interactive mode"
It forces the user to create an admin user name that is not equal to cluster-admin under the htpasswd idp that is named "cluster-admin".
(IDP NAME) image

(USERNAME) image

To summarize, `rosa describe admin` (before fix introduced in this PR) works if the user creates an admin via "rosa create admin" , because both the idp name and username is "cluster-admin".

If the username associated under the htpasswd idp named "cluster-admin" adopts a username that is different from the idp name (as enforced by rosa create cluster --cluster-admin-user), the  `rosa describe admin` (before fix introduced in this PR)  will not work.

In addition to command `rosa describe admin`, it also impacts `rosa delete admin` command as well.

A decision is required to sync up on whether the initial admin username (not IDP name) elected to be created, will always be "cluster-admin" or something else. @ciaranRoche @oriAdler @gdbranco @marcolan018

**Comment by ckandag** [comment]
> A decision is required to sync up on whether the initial admin username (not IDP name) elected to be created, will always be "cluster-admin" or something else. @ciaranRoche @oriAdler @gdbranco @marcolan018


Looking at the past history  `rosa create admin` always enforced that admin username == "cluster-admin". 
The main change I did was to include this cluster-admin user in an IDP also named "cluster-admin" to differentiate it from other Idps ,  previously  this admin IDP name could have been anything and defaulted simply to "htpasswd".


In the `rosa cluster create --admin-user=` ,     While I worked with Marco to ensure this create path similarly  used  IDP = `cluster-admin`.  Looks like we are not enforcing  user-name==cluster-admin.

Fixing cluster-creation code path  also to always have  "user-name==cluster-admin" and not allow any custom user-names would keep it consistent across ROSA and will automatically.

**Comment by davidleerh** [comment]
PR closed in favor of: https://github.com/openshift/rosa/pull/1522

**Pull request closed without merging**